pkg/config/validation/validation.go

```diff
@@ -1936,6 +1937,27 @@ func ValidateControlPlaneAuthPolicy(policy meshconfig.AuthenticationPolicy) erro
 	return fmt.Errorf("unrecognized control plane auth policy %q", policy)
 }
 
+func validatePolicyTargetReference(selector *type_beta.WorkloadSelector, targetRef *type_beta.PolicyTargetReference) (v Validation) {
+	if targetRef == nil {
+		return
+	}
+	if selector != nil && targetRef != nil {
+		v = appendErrorf(v, "targetRef and selector cannot both be set")
+	}
+	if targetRef.Name == "" {
+		v = appendErrorf(v, "targetRef name must be set")
+	}
+	if targetRef.Namespace != "" {
+		v = appendErrorf(v, "targetRef namespace must not be set")
```

Comment:
Not sure if namespace cannot be set, how could we apply the policy to a gateway, as previously the policy can be in caller's namespace

Reply:
And if we allow cross namespace reference, there is another security issue.

Reply:
@keithmattix didn't we want to not set the namespace?

Reply:
Correct, here's @howardjohn's comment on it: https://github.com/istio/api/pull/2888#discussion_r1316219986

Basically, even for ingress Gateways, policies must be colocared with the gateway resource right now

Reply:
> And if we allow cross namespace reference, there is another security issue

Cross namespace shouldn't be allowed. If you look at the gateway api spec for[ direct policy attachment](https://gateway-api.sigs.k8s.io/geps/gep-713/#direct-policy-attachment), it states that "Direct Policy Attachment should only be used to target objects in the same namespace as the Policy object. Allowing cross-namespace references brings in significant security concerns..." and in the Istio context John points out the same thing in a comment on the API[ proto PR](https://github.com/istio/api/pull/2888#discussion_r1298759265) that just merged.

Reply:
@hzxuzhonghu, here's the comment where we decided to invalidate any targetRef that decides to set the namespace: https://github.com/istio/api/pull/2888#discussion_r1316219986

As of now, all usages of targetRef will be using the same namespace. We may need the namespace field in the future, but not now.

So I think we should stick to not allowing namespace to be set.

Reply:
the targetRef.Namespace can be set, but it must be equal to the object itself. 

Yes, if future cross namespace support needed, we have to use referenceGrant or similar

Reply:
Actually, one way we can make sure there's no cross namespace references is by not allowing users to set the namespace at all. Not setting a namespace for targetRef means the target object must be in the policy's namespace (which is what we want until we support referenceGrant)